src/components/Consent.js

```diff
@@ -1,44 +1,46 @@
 import { h } from 'preact'
-import { useI18n, useSoftkey, useOnlineStatus } from 'hooks'
+import { useRef, useEffect } from 'preact/hooks'
+import { useI18n, useSoftkey, useOnlineStatus, useScroll } from 'hooks'
 import { grantConsent, goto } from 'utils'
-import { OfflinePanel } from 'components'
 
-export const Consent = () => {
+export const Consent = ({ close }) => {
   const i18n = useI18n()
   const isOnline = useOnlineStatus()
+  const containerRef = useRef()
+  const [scrollDown, scrollUp] = useScroll(containerRef, 10, 'y')
 
   const onAgree = () => {
     grantConsent()
-    goto.search()
+    close()
   }
 
-  const softkeyConfig = {
-    offline: {
-      backspace: () => { window.exit() }
-    },
-    online: {
-      center: i18n('consent-softkeys-agree'),
-      onKeyCenter: onAgree,
-      left: i18n('consent-softkeys-terms'),
-      onKeyLeft: goto.termsOfUse,
-      right: i18n('consent-softkeys-policy'),
-      onKeyRight: goto.privacyPolicy,
-      backspace: () => { window.exit() }
+  useSoftkey('ConsentMessage', {
+    center: i18n('consent-softkeys-agree'),
+    onKeyCenter: onAgree,
+    left: i18n('consent-softkeys-terms'),
+    onKeyLeft: goto.termsOfUse,
+    right: i18n('consent-softkeys-policy'),
+    onKeyRight: goto.privacyPolicy,
+    onKeyArrowDown: scrollDown,
+    onKeyArrowUp: scrollUp,
+    onKeyBackspace: () => { window.close() }
+  })
+
+  useEffect(() => {
+    if (!isOnline) {
+      close()
     }
-  }
-  useSoftkey('ConsentMessage', softkeyConfig[isOnline ? 'online' : 'offline'], [isOnline], true)
+  }, [isOnline])
```

Comment:
This logic looks weird and doesn't make sense and the controller of the online/offline state is in the upper level (search screen)

I propose `usePopup` to return `close` method, then within the search component we use

```js
useEffect(() => {
    // ... other code
    if (!isOnline) {
      closeConsentPopup()
    }
    // ... other code
  }, [isOnline])
```